e2e/testcafe-devextreme/tests/dataGrid/filterRow/filterRow.ts

```diff
@@ -9,6 +9,49 @@ import { getNumberData } from '../helpers/generateDataSourceData';
 fixture.disablePageReloads`FilterRow`
   .page(url(__dirname, '../../container.html'));
 
+test('Filter should reset if the filter row editor text is cleared (T1257261)', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const filterEditor = dataGrid.getFilterEditor(1, FilterTextBox);
+  const filterPanelText = dataGrid.getFilterPanel().getFilterText();
+
+  await t
+    // assert
+    .expect(filterPanelText.element.textContent)
+    .eql('[Text] Equals \'i\'')
+    // act
+    .click(filterEditor.input)
+    .pressKey('backspace')
+    .wait(100) // updateValueTimeout
+    // assert
+    .expect(filterPanelText.element.textContent)
+    .eql('Create Filter')
+    // act
+    .click(dataGrid.element)
+    // assert
+    .expect(filterPanelText.element.textContent)
+    .eql('Create Filter');
+}).before(async () => createWidget('dxDataGrid', {
+  dataSource: [
+    { ID: 1, Text: 'Item 1' },
+    { ID: 2, Text: '' },
+    { ID: 3, Text: 'Item 3' },
+  ],
+  keyExpr: 'ID',
+  showBorders: true,
+  remoteOperations: true,
+  headerFilter: { visible: true },
+  filterRow: { visible: true },
+  filterPanel: { visible: true },
+  filterValue: ['Text', '=', 'i'],
+  columns: ['ID', {
+    dataField: 'Text',
+    selectedFilterOperation: '=',
+  }],
+  onEditorPreparing(e: any) {
+    e.updateValueTimeout = 100;
```

Comment:
It's related to an issue I encountered with my initial solution. The use case is as follows:
1) User input in the filter row.
2) User clears the input via the backspace key.

There's a delay/timeout at this point before the new input is registered. After the timeout, the grid's filter is cleared.

3) User clicked outside the filter row editor (here's the reason why there's the last click). The focus is removed from the editor (blur event).

Since the editor's default `valueChangeEvent` includes 'change', the blur event triggers the editor's value change event. Unfortunately, at this point, the editor re-applies its old value.

This test case ensures that the same filter applied via input persists even after the 'change' event is triggered.